.ci/jenkins/Jenkinsfile.publish

```
@@ -0,0 +1,97 @@
+@Library('jenkins-pipeline-shared-libraries')_
+import org.kie.jenkins.MavenCommand
+
+AGENT_LABEL="kie-rhel7 && kie-mem4"
+MVN_TOOL="kie-maven-3.8.1"
+JDK_TOOL="kie-jdk11"
+REPO="kiegroup-website"
+ORGANIZATION="kiegroup"
+BASE_BRANCH="main"
+MAIL_RECIPIENT="jporter@redhat.com"
+
+pipeline {
+    agent {
+        label "$AGENT_LABEL"
+    }
+    options{
+        timestamps()
+    }
+    tools {
+        maven "$MVN_TOOL"
+        jdk "$JDK_TOOL"
+    }
+    stages {
+        stage('CleanWorkspace') {
+            steps {
+                cleanWs()
+            }
+        }
+        stage('clone kiegroup-website') {
+            steps {
+                script {
+                    checkoutStartkiegroupWeb()
+                }
+            }
+        }
+        stage('Front end compilation') {
+            steps {
+                dir("${REPO}") {
+                    sh 'echo ${NODE_EXTRA_CA_CERTS}'
+                    sh 'npm install --registry=${NPM_REGISTRY_URL} && npm run-script build:webpack'
```

```suggestion
                    sh 'npm exec @kie/lock-treatment-tool -- --registry=${NPM_REGISTRY_URL} && npm install --registry=${NPM_REGISTRY_URL} && npm run-script build:webpack'
```
The resolved fields from [yarn.lock](https://github.com/kiegroup/kiegroup-website/blob/6f528f81dc9d1a3ba6ff24bba9465e50ab418e6d/yarn.lock#L7) or [package-lock-json](https://github.com/kiegroup/kiegroup-website/blob/6f528f81dc9d1a3ba6ff24bba9465e50ab418e6d/package-lock.json#L7) files should be treated. I recommend to use https://www.npmjs.com/package/@kie/lock-treatment-tool for this. Otherwise the lock files will be a problem behind the proxy.
It is very easy to use/add by just executing `npm exec @kie/lock-treatment-tool -- --registry=${NPM_REGISTRY_URL}` before installing the project.

---

@LightGuard I see we have package-lock.json and yarn.lock files. I guess we can remove the package-lock.json file, right? Can you please confirm? If this is the case we should use `yarn` commands instead of `npm` ones
```suggestion
                    sh 'npm exec @kie/lock-treatment-tool -- --registry=${NPM_REGISTRY_URL} && yarn install --registry=${NPM_REGISTRY_URL} && yarn build:webpack'
```

---

I keep this confirmation open for @LightGuard to confirm